tests/testthat/test-infrastructure.r

```diff
@@ -59,3 +59,96 @@ test_that("use_data", {
     expect_equal(local_test_data_item_to_save, 42L)
   })
 })
+
+test_that("add_desc_package only modifies 'field' section of DESCRIPTION", {
+  # use tests/testthat/testData as testing package
+  path_to_test_desc <- file.path("testData", "DESCRIPTION")
+  old_desc <- read_dcf(path_to_test_desc)
+
+  # be sure to return the description to original state on exit
+  on.exit(write_dcf(path_to_test_desc, old_desc))
+
+  # add testPackage to the "Imports" of the testData package
+  import_text <- "testPackage (>= 1.0.0)"
+  output <- add_desc_package("testData", "Imports", import_text)
+  new_desc <- read_dcf(path_to_test_desc)
+
+  expect_true(output) # check that add_desc_package still returns boolean
+  expect_equal(old_desc, new_desc[which(names(new_desc) != "Imports")]) # make sure only Imports section is modified
+  expect_equal(new_desc[["Imports"]], import_text) # make sure Import section is properly modified
+})
+
+test_that("add_desc_package is idempotent", {
+  # use tests/testthat/testData as testing package
+  path_to_test_desc <- file.path("testData", "DESCRIPTION")
+  old_desc <- read_dcf(path_to_test_desc)
+
+  # be sure to return the description to original state on exit
+  on.exit(write_dcf(path_to_test_desc, old_desc))
+
+  # add testPackage to the Imports: of the testData package;
+  # do it twice and check that the description file is the same as in
+  # the second time.
+  import_text <- "testPackage (>= 1.0.0)"
+  descriptions <- lapply(c(1,2), function(call_repetition) {
+    if (call_repetition == 1) {
+      # first time function is being called... It should return TRUE.
+      expect_true(add_desc_package("testData", "Imports", import_text))
+    } else {
+      # function should report that nothing changed
+      expect_false(add_desc_package("testData", "Imports", import_text))
+    }
+
+    return(read_dcf(path_to_test_desc))
+    })
+
+  # finally, check for idempotency.
+  expect_equal(descriptions[1], descriptions[2])
+})
+
+test_that("use_package throws error with invalid 'version' argument", {
+  # test error is thrown with invalid `version`;
+  # use the tests/testthat/testData/ as test package.
+  expect_error(use_package("utils", "Imports", "testData", "invalid"))
+})
+
+# Set up a grid of the potential inputs to use_package and individual test
+# them for correctness and behaviour. These are only partial tests, since the
+# checks for idempotency, error-throwing, and making sure *only* the indicated
+# fields are affected, are all handled above.
+test_that("use_package can modify Imports, Suggests and Depends", {
```

Comment:
I think you can drop this entire test - 30 lines of code seems overkill to test a single line in the function. You're effectively just testing that `match.arg()` does what it's supposed to.